gtfs_rt_validator_api.py

```diff
@@ -215,62 +270,99 @@ def validate_gcs_bucket_many(
+        params = params.iloc[:limit]
+
     # check that the parameters file has all required columns
     missing_cols = set(required_cols) - set(params.columns)
     if missing_cols:
         raise ValueError("parameter csv missing columns: %s" % missing_cols)
 
-    status = []
-    for idx, row in params.iterrows():
-        try:
-            validate_gcs_bucket(
+    statuses = []
+
+    logger.info(f"processing {params.shape[0]} inputs with {threads} threads")
+
+    # https://github.com/fsspec/gcsfs/issues/379#issuecomment-826887228
+    # Note that this seems to differ per OS
+    ctx = multiprocessing.get_context("spawn")
+
+    # from https://stackoverflow.com/a/55149491
+    # could be cleaned up a bit with a namedtuple
+    with ProcessPoolExecutor(max_workers=threads, mp_context=ctx) as pool:
+        futures = {
+            pool.submit(
+                validate_gcs_bucket,
                 project_id,
                 token,
-                results_bucket=results_bucket + f"/{result_name_prefix}{idx}.parquet",
                 verbose=verbose,
+                results_bucket=results_bucket + f"/{result_name_prefix}{idx}.parquet",
                 aggregate_counts=aggregate_counts,
-                **row[required_cols]
-            )
-
-            status.append({**row, "is_success": True})
-        except Exception as e:
-            if strict:
-                raise e
-
-            status.append({**row, "is_success": False})
-    
-    status_newline_json = "\n".join([json.dumps(record) for record in status])
+                idx=idx,
+                **row[required_cols],
+            ): row
+            for idx, row in params.iterrows()
+        }
+
+        # Processes each future as it is completed, i.e. returned or errored
+        for future in concurrent.futures.as_completed(futures):
+            row = futures[future]
+            # result() will throw an exception if one occurred in the underlying function
+            try:
+                future.result()
+            except Exception as e:
+                if strict:
+                    raise e
+                statuses.append({**row, "is_success": False})
+            else:
+                statuses.append({**row, "is_success": True})
+
+    successes = sum(s["is_success"] for s in statuses)
+
+    logger.info(f"finished multiprocessing; {successes} successful of {len(statuses)}")
+
+    status_newline_json = "\n".join([json.dumps(record) for record in statuses])
 
     if status_result_path:
-        fs.pipe(status_result_path, status_newline_json.encode()) 
+        fs.pipe(status_result_path, status_newline_json.encode())
 
 
 def download_gtfs_schedule_zip(gtfs_schedule_path, dst_path, fs):
     # fetch and zip gtfs schedule
+    logger.info(f"Fetching gtfs schedule data from {gtfs_schedule_path} to {dst_path}")
     fs.get(gtfs_schedule_path, dst_path, recursive=True)
+    try:
+        os.remove(os.path.join(dst_path, "areas.txt"))
```

Comment:
Did we see any errors for other optional files? (Does this run successfully on *all* feeds with this line included?)

Reply:
I haven't seen any so far; will see if I can execute on everything before I merge.

Reply:
I was able to execute a fair number and didn't hit a similar error; lots of missing RT data though

`finished multiprocessing; 208 successful of 651`